units/event_log_tests.cc

```diff
@@ -205,6 +208,115 @@ std::string EventToCSVString(Event &event) {
   return csv_str.str();
 }
 
+// EventLogger base class
+//
+// EventLogger provides an abstraction for the concrete loggers that log the
+// events in various formats (CSV, Chrome Trace Event, etc).
+class EventLogger {
+ public:
+  virtual ~EventLogger() = default;
+
+  // Converts an `Event` into a string and writes it to the output.
+  virtual void AddEvent(Event *event) = 0;
+
+  // This method is called exactly once before any events are added to denote
+  // that the log has started. For instance, in case of a CSV log, the CSV
+  // header is written when this function is called.
+  virtual void StartLog() = 0;
+
+  // This method is called exactly once to denote that the log is finished.
+  virtual void EndLog() = 0;
+
+  // Makes sure all the logs are written to the output.
+  virtual void Flush() = 0;
+};
+
+// FilterLogger is a wrapper around the `EventLogger` that filters the input
+// events based on a given log level.
+// Example:
+// ```C++
+// CSVLogger logger = ...;
+// FilterLogger(&logger, LogLevel::High);
+// ```
+class FilterLogger : public EventLogger {
+ public:
+  FilterLogger(EventLogger *logger, LogLevel log_level)
+      : logger_(logger), log_level_(log_level){};
+
+  void AddEvent(Event *event) override {
+    if (event->GetLogLevel() >= log_level_) logger_->AddEvent(event);
+  }
+
+  void StartLog() override { logger_->StartLog(); }
+
+  void EndLog() override { logger_->EndLog(); }
+
+  void Flush() override { logger_->Flush(); }
+
+ private:
+  EventLogger *logger_ = nullptr;
+  LogLevel log_level_ = LogLevel::Low;
+};
+
+// A subclass of `EventLogger` that forwards all events, start/end, and flushes
+// to all its children. Example: `BroadCastLogger({&event_logger1,
+// &event_logger2});`
```

Comment:
nit: put this example in a code block, it's a bit difficult to read when it spans 2 lines like this